src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -46,7 +46,7 @@ function buildEcdhParam(public_algo, oid, cipher_algo, hash_algo, fingerprint) {
     new Uint8Array([public_algo]),
     kdf_params.write(),
     util.str_to_Uint8Array("Anonymous Sender    "),
-    fingerprint
+    fingerprint.subarray(0, 20)
```

Comment:
Wait, is this supposed to be the first 20 bytes of the fingerprint or the first 10 bytes? If the input is a hex string here then this takes the first 10 bytes. Right?

Reply:
`fingerprint` is an `Uint8Array` here.